src/components/inputs/date-input/date-input.js

```diff
@@ -158,7 +158,6 @@ const DateInput = props => {
                     typeof props.placeholder === 'string'
                       ? props.placeholder
                       : intl.formatMessage(messages.placeholder),
-                  disabled: props.isDisabled,
```

Comment:
We're already passing the `isDisabled` prop to CalendarBody. By also passing the `disabled` state here and directly to the input, it messes with the correct order of the states and the styles that should prevail.